apps/docker/entrypoint.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Bash strict mode
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+
+# Begin env variables for configuration
+#
+# Config for this script. Only matters if you're using SAFE_WORLDSERVER
+SAFE_WORLDSERVER="${SAFE_WORLDSERVER:-}"
+ACORE_SKIP_CREATE_ACCOUNT="${ACORE_SKIP_CREATE_ACCOUNT:-}"
+ACORE_GM_LEVEL="${ACORE_GM_LEVEL:-3}"
+ACORE_USERNAME="${ACORE_USERNAME:-admin}"
+ACORE_PASSWORD="${ACORE_PASSWORD:-admin}"
+grep -qE  "[^a-zA-Z0-9]" <<< "$ACORE_PASSWORD" && \
+  echo "ACORE_PASSWORD is not alphanumeric. ACORE_PASSWORD should only be letters and numbers. Exiting..." && \
+  exit 1
+
+# Config for the application itself
+
+export MYSQL_HOST="${MYSQL_HOST:-ac-database}"
+export MYSQL_USER="${MYSQL_USER:-root}"
+export MYSQL_PASSWORD="${MYSQL_PASSWORD:-azerothcore}"
+export MYSQL_PORT="${MYSQL_PORT:-3306}"
+# End env variables for configuration
+
+CMD="${CMD:=$@}"
+
+# Template env vars into configs
+
+cp -rvn /azerothcore/env/ref/etc/* /azerothcore/env/dist/etc
+
+conf="/azerothcore/env/dist/etc/$ACORE_COMPONENT.conf" 
+
+[ -f "$conf" ] && \
+  # Substitute the configuration with environment variables
+  envsubst < "$conf" > "$conf.new" && \
+  mv "$conf.new" "$conf"
+
+if [ -n "$SAFE_WORLDSERVER" ] && \
+   [ "$ACORE_COMPONENT" == "worldserver" ] && \
+   grep -q "$ACORE_COMPONENT" <<< "$CMD" ; then
+  cat << EOF 
+#==============================================================================#
+# You are using SAFE_WORLDSERVER mode.
+#
+# Be aware:
+#   - You CANNOT use the worldserver console.
+#   - All commands must be performed over SOAP or telnet.
+#
+# Please see [ https://www.azerothcore.org/wiki/remote-access ] for further info.
+#==============================================================================#
```

Comment:
If we do not allow to use the console, these guides are not going to be valid anymore:

Acore project: https://github.com/azerothcore/acore-docker#access-the-worlserver-console-and-create-an-account

Access the worldserver: https://www.azerothcore.org/wiki/install-with-docker

Would it be possible to offer this as an option for those that really need to have the docker images in a safe production environment?

Reply:
Yeah, I'm conflicted on it. Personally, I think that the console in the context of docker can counteract the benefits of docker, but as it stands the console is an essential component.

Updating documentation is an easy fix, however. 

I can make a small script that works like a console so there's transparent functionality to using the repl. Python's stdlib has a [Cmd module](https://docs.python.org/3/library/cmd.html) that'd probably make this pretty easy.

Reply:
> Would it be possible to offer this as an option for those that really need to have the docker images in a safe production environment?

yeah that would be nice, I'd prefer to keep this possibility

Reply:
@michaeldelago we have a `acore-docker` project (https://github.com/azerothcore/acore-docker)  that is intended to be a simple docker-compose, that uses the images deployed by the azerothcore-wotlk repo, to spin up the azerothcore without knowing anything else than how to install docker. 

The goal is to keep the testing of azerothcore simple and stupid for those people that want to give AC a go with zero-knowledge or even for bug triagers etc.
These people do not want to follow complex steps to setup a telnet connection and just want to attach their terminal to the container console. 

That project is cloned 30 times per day and is pretty essential for us so we need to support this ability keeping the documentation untouched or slightly changed without increase the complexity.

Reply:
To clarify, the behavior that we're talking about is optional and opt-in. The images that are generated work in the way the documentation describes, though there is an extra mode for running the worldserver that has some properties that some people may find to be useful.

The lines referenced that shows a brief warning when using `SAFE_WORLDSERVER` mode. `SAFE_WORLDSERVER` mode controls the worldserver through an `expect` script, as well as creates an admin account for the server. If the container receives a SIGTERM, this expect script will issue a `server shutdown 5s 0` command to the worldserver. 

`SAFE_WORLDSERVER` mode is only activated if the operator has set the `SAFE_WORLDSERVER` environment variable, seen [in the entrypoint](https://github.com/michaeldelago/azerothcore-wotlk/blob/docker-multi-stage/apps/docker/entrypoint.sh#L10) and [the compose file](https://github.com/michaeldelago/azerothcore-wotlk/blob/docker-multi-stage/docker-compose.yml). Without `SAFE_WORLDSERVER`, the worldserver will operate in the exact same way that many people are used to.

I will note that I did just remove the default setting of "true" in the compose file, though that's doesn't change the operations for https://github.com/azerothcore/acore-docker

Reply:
ah perfect then, if you need to enable the SAFE_WORLDSERVER then we can resolve this conversation. Can you also prepare the PR to update our wiki please?

Reply:
Yeah, I can do that